### PR TITLE
libpqxx: adding Makefile, Config.in and patch

### PR DESCRIPTION
**Pull request by garikello3d**
Signed-off-by: Igor Bezzubchenko <garikello@gmail.com>

Maintainer: me / @garikello3d

Compile tested - based on OpenWRT revision 7ba2f5c96f2697e8 (28 Nov 2020)
- QEMU x86_64
- QEMU ARM Cortex-a15 (_-M virt_)
- QEMU MIPS (_-M malta_)

Run tested: the same as compile

Description: I'm adding the official C++ Postgres Client library (official [web](http://pqxx.org/development/libpqxx/l)).

My build and functional test includes successful run of full test suite available in the project against a host-based PostgreSQL server.

### CODE REVIEW

File: libs/libpqxx/Makefile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=libpqxx
+PKG_VERSION:=7.3.0
+PKG_RELEASE:=1
+PKG_INSTALL:=1
```
Comment (neheb) [inline libs/libpqxx/Makefile:6]: cmake.mk makes this default. Remove.

File: libs/libpqxx/Makefile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	SUBMENU:=Database
+endef
+
+define Package/libpqxx/config
+	source "$(SOURCE)/Config.in"
+endef
+
+CMAKE_OPTIONS += \
+	-DBUILD_DOC=off \
+	$(if $(CONFIG_LIBPQXX_STATIC), -DBUILD_STATIC_LIBS=on) \
```
Comment (neheb) [inline libs/libpqxx/Makefile:35]: I'd rather see something like

-DBUILD_STATIC_LIBS=O$(if $(CONFIG_LIBPQXX_STATIC),N, FF)

same as below.

File: libs/libpqxx/Makefile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	URL:=http://pqxx.org/development/libpqxx
+	SUBMENU:=Database
+endef
+
+define Package/libpqxx/config
+	source "$(SOURCE)/Config.in"
+endef
+
+CMAKE_OPTIONS += \
+	-DBUILD_DOC=off \
```
Comment (neheb) [inline libs/libpqxx/Makefile:34]: Please capitalize options. That's how it's done elsewhere.

File: libs/libpqxx/Makefile [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	$(if $(CONFIG_LIBPQXX_STATIC), -DBUILD_STATIC_LIBS=on) \
+	$(if $(CONFIG_LIBPQXX_SHARED), -DBUILD_SHARED_LIBS=on)
+
+ifeq ($(CONFIG_LIBPQXX_INSTALL_TEST),y)
```
Comment (neheb) [inline libs/libpqxx/Makefile:38]: There's no reason for this. Just make them a part of the main CMAKE_OPTIONS.

File: libs/libpqxx/Makefile [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+PKG_VERSION:=7.3.0
+PKG_RELEASE:=1
+PKG_INSTALL:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/jtv/libpqxx
+PKG_SOURCE_VERSION:=$(PKG_VERSION)
+PKG_MIRROR_HASH:=c176771dbbf02e38b84a8f9bf986a83587e2d86d9e0d2238e2a17a036e865951
+
+CMAKE_INSTALL:=1
```
Comment (neheb) [inline libs/libpqxx/Makefile:13]: One issue I've had with CMAKE_INSTALL in the past is that the pkgconfig paths are totally wrong. See https://github.com/openwrt/packages/blob/master/libs/pugixml/Makefile#L45 for an example.

Please check that libdir and includedir are correct.

File: libs/libpqxx/Makefile (hunk 1 above)
Comment (garikello3d) [reply libs/libpqxx/Makefile]: Fixed

File: libs/libpqxx/Makefile (hunk 2 above)
Comment (garikello3d) [reply libs/libpqxx/Makefile]: Fixed

File: libs/libpqxx/Makefile (hunk 3 above)
Comment (garikello3d) [reply libs/libpqxx/Makefile]: Fixed

File: libs/libpqxx/Makefile (hunk 4 above)
Comment (garikello3d) [reply libs/libpqxx/Makefile]: Fixed

File: libs/libpqxx/Makefile (hunk 5 above)
Comment (garikello3d) [reply libs/libpqxx/Makefile:12]: I checked the **libpqxx.pc** file generated.

> prefix=/usr
> exec_prefix=${prefix}
> libdir=${prefix}/lib
> includedir=${prefix}/include

So libdir and include dir are parameterized by _prefix_, nothing bad that I see...

**Pull request merged**